moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -107,28 +107,15 @@ void plan_execution::PlanExecution::stop()
 
 void plan_execution::PlanExecution::planAndExecute(ExecutableMotionPlan& plan, const Options& opt)
 {
-  plan.planning_scene_monitor_ = planning_scene_monitor_;
-  plan.planning_scene_ = planning_scene_monitor_->getPlanningScene();
-  planAndExecuteHelper(plan, opt);
+  planAndExecute(plan, moveit_msgs::PlanningScene(), opt);
 }
 
 void plan_execution::PlanExecution::planAndExecute(ExecutableMotionPlan& plan,
                                                    const moveit_msgs::PlanningScene& scene_diff, const Options& opt)
 {
-  if (moveit::core::isEmpty(scene_diff))
-    planAndExecute(plan, opt);
-  else
-  {
-    plan.planning_scene_monitor_ = planning_scene_monitor_;
-    {
-      planning_scene_monitor::LockedPlanningSceneRO lscene(planning_scene_monitor_);  // lock the scene so that it does
-                                                                                      // not modify the world
-                                                                                      // representation while diff() is
-                                                                                      // called
-      plan.planning_scene_ = lscene->diff(scene_diff);
-    }
-    planAndExecuteHelper(plan, opt);
-  }
+  plan.planning_scene_monitor_ = planning_scene_monitor_;
+  plan.planning_scene_ = planning_scene_monitor_->copyPlanningScene(scene_diff);
+  planAndExecute(plan, opt);
```

Comment:
Shouldn't this call `planAndExecuteHelper`? Otherwise, both methods call each other recursively forever.
```suggestion
  planAndExecuteHelper(plan, opt);
```